app/Http/Controllers/BookingController.php

```diff
@@ -70,15 +70,30 @@ public function create(Request $request)
 
     }
 
-    public function all()
+    public function all(Request $request)
     {
-        $hidden = ['guests', 'updated_at'];
+        $search = $request->input('room_id');
+        $hidden = ['guests', 'room_id', 'booking_id', 'guests', 'updated_at'];
         $date = today()->toDateString();
+        $data = Booking::with(['rooms:id,name'])->whereRelation('rooms', 'room_id', '=', "$search")
```

Comment:
You should move this into the if ($search) if statement, only create this set of results if there is a search